### PR TITLE
reflect vrrp state in system overview

### PR DESCRIPTION
**Pull request by didovesei**

### CODE REVIEW

File: forch/forchestrator.py [hunk 1]
```diff
@@ -192,6 +191,10 @@ def _get_combined_summary(self, summary):
             detail = 'broken subsystems: ' + ', '.join(details)
         else:
             detail = 'n/a'
+
+        vrrp_state = self._local_collector.get_vrrp_state()
+        if not vrrp_state.get('is_master'):
+            return constants.STATE_INACTIVE, detail
```
Comment (grafnu) [inline forch/forchestrator.py]: For this the detail should be something like "Controller is inactive, please view peer controller."

File: forch/local_state_collector.py [hunk 2]
```diff
@@ -40,10 +44,17 @@ def get_process_summary(self):
...
         """Get the states of processes"""
         process_state = None
-        with self._process_lock:
+        with self._lock:
             process_state = copy.deepcopy(self._process_state)
         return process_state
 
+    def get_vrrp_state(self):
+        """Get the local VRRP state"""
+        vrrp_state = None
```
Comment (grafnu) [inline forch/local_state_collector.py]: This line has no effect (plz remove)

File: forch/local_state_collector.py [hunk 3]
```diff
@@ -40,10 +44,17 @@ def get_process_summary(self):
...
+        with self._lock:
+            vrrp_state = copy.deepcopy(self._vrrp_state)
```
Comment (grafnu) [inline forch/local_state_collector.py]: just return directly (no need for intermediate variable)

Comment (didovesei) [reply forch/local_state_collector.py]: I think Python returns a reference to the map. I don't think "deepcopy" is not thread safe as well... Do we need to consider a race condition here?

Comment (didovesei) [reply forch/local_state_collector.py]: Do you mean:
with self._lock:
    return copy.deepcopy(...)
?

File: forch/forchestrator.py (hunk 1 above)
Comment (didovesei) [reply forch/forchestrator.py]: Done

File: forch/local_state_collector.py (hunk 3 above)
Comment (grafnu) [reply forch/local_state_collector.py]: yes -- the deepcopy is fine, it's just the "vrrp_state" variable that's unnecessary

Comment (didovesei) [reply forch/local_state_collector.py]: Got it! Didn't know it is possible to do a return under "with" statement... PTAL

File: forch/local_state_collector.py (hunk 2 above)
Comment (didovesei) [reply forch/local_state_collector.py]: Done

**Comment by didovesei** [comment]
> Code looks good, just need to make it configurable. I'd recommend a "check_keepalived" entry somewhere, defaults to False, and then we can set it to True in the configuration. I think it's close enough to processes if you want to just include it under there somewhere.

Sure

**Pull request merged**